### DISCUSSION TITLE
trying to create a custom tool but error on chat: Received tool input did not match expected schema

### DISCUSSION THREAD

**Discussion by remixdance** [FlowiseAI/Flowise#1082] [Q&A]
Hi im trying to create a custom tool with an agent conversation here the code: 

var myHeaders = new Headers();
myHeaders.append("X-API-Key", "MYAPI");

var requestOptions = {
  method: 'GET',
  headers: myHeaders,
  redirect: 'follow'
};

fetch("https://api.syntage.com/v2/insights/DASD1231/balance-sheet", requestOptions)
  .then(response => response.text())
  .then(result => console.log(result))
  .catch(error => console.log('error', error));

when i save and do a question the bot answerd : Received tool input did not match expected schema here my flow : 
![image](https://github.com/FlowiseAI/Flowise/assets/42038961/36d321da-584a-4afc-b133-c00325813396)

**Comment by HenryHengZJ** [comment]
Try using OpenAI Function Agent

**Comment by asimgilani** [comment]
Having a similar issue. My tool has 10 optional input params. OpenAI function agent may at any given point utilize any number of those 10 optional inputs. 

Lets say if it only utilizes 2 of the optional inputs, it will throw the error: "Received tool input did not match expected schema"

Sometimes randomly the OpenAI function will utilize 2 out of 10 inputs but still include the remaining 8 with empty values, in that case it works fine.

In short, if the number of param input auto suggested by OpenAI function does not match what's been created in the tool, it will throw this error. 

Is there any way around this?
